distributed/comm/ucx.py

```diff
@@ -35,6 +35,15 @@
 cuda_array = None
 
 
+def synchronize_stream(stream=0):
+    import numba.cuda
+
+    ctx = numba.cuda.current_context()
```

Comment:
Wow, that's a lot of work to synchronize a stream you already have.... Is this necessary? Can you just call the CUDA driver API directly?

Reply:
That's probably more complex than this change. Reason being we need the logic to find that library and load it. Numba is helping us bypass that step.

Reply:
In Numba 0.49.0, there's https://numba.pydata.org/numba-doc/dev/cuda-reference/host.html#numba.cuda.default_stream, but not on earlier versions unfortunately. So the code here is basically doing what the new `numba.cuda.default_stream` function does, plus synchronizing that stream.

How would I call the CUDA driver API directly from Dask with less work than that, and also not depending on some more lower-level libraries? Do you know of a way to do so?

Reply:
> How would I call the CUDA driver API directly from Dask with less work than that, and also not depending on some more lower-level libraries? Do you know of a way to do so?

You could technically do it via ctypes but then you'd need to reimplement the status code checking and whatnot where it's absolutely not worth it.

If this has problematic performance we can add something in rmm Python to deal with it.

Reply:
It's also usually a chore getting ctypes to load the right library. Agree it would be better to avoid ctypes if possible.

Reply:
Yeah, in that case it sounds more convoluted than what's here currently. It doesn't seem problematic performance-wise as of now, so I would prefer that we get this in to unblock people asap, and do such improvements later if they're really needed.

Reply:
OK, I was just asking.